File: hmda/src/main/scala/hmda/model/institution/Parent.scala
package hmda.model.institution

object Parent {
  def empty: Parent = Parent(None, None)
}

case class Parent(
    idRssd: Option[Int],
    name: Option[String]
) {
  def isEmpty: Boolean = {
    this match {
      case Parent(None, None) => true
      case _                  => false
    }
  }
}

File: kubernetes/keycloak/auth/src/main/java/gov/cfpb/keycloak/authenticator/hmda/Parent.java
package gov.cfpb.keycloak.authenticator.hmda;

public class Parent {

    private Integer idRssd;
    private String name;

    public Parent() {
    }

    public Integer getIdRssd() {
        return idRssd;
    }

    public void setIdRssd(Integer idRssd) {
        this.idRssd = idRssd;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public String toString() {
        return "Parent{" +
                "idRssd=" + idRssd +
                ", name='" + name + '\'' +
                '}';
    }
}
